lib/cloudant.js

```diff
@@ -580,6 +590,59 @@ Cloudant.prototype.ping = function(cb) {
 };
 
 /**
+ * Replace if the model instance exists with the same id or create a
+ * new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [cb] The callback function
+ */
+Cloudant.prototype.replaceOrCreate = function replaceOrCreate(
+  model, data, options, cb) {
+  debug('Cloudant.prototype.replaceOrCreate %j %j', model, data);
+  var self = this;
+  var idName = self.idName(model);
+  var id = data[idName];
+  if (id) {
+    self.getCurrentRevision(model, id, function(err, rev) {
+      if (err) return cb(err);
+      if (rev) data._rev = rev;
+      self.create(model, data, options, function(err) {
+        if (err) return cb(err);
+        self.findById(model, data.id, {}, function(err, result) {
+          if (err) return cb(err);
+          cb(err, result, {isNewInstance: !rev});
+        });
+      });
+    });
+  } else {
+    self.create(model, data, options, function(err, id) {
+      if (err) return cb(err);
+      self.findById(model, data.id, {}, function(err, result) {
+        if (err) return cb(err);
+        cb(err, result, {isNewInstance: true});
+      });
+    });
+  }
+};
+
+/**
+ * Replace properties for the model instance data
+ * @param {String} model The name of the model
+ * @param {*} id The instance id
+ * @param {Object} data The model data
+ * @param {Object} options The options object
+ * @param {Function} [cb] The callback function
+ */
+
+Cloudant.prototype.replaceById = function replaceById(
+  model, id, data, options, cb) {
+  data.id = id,
+  options.upsert = false;
+  this.replaceOrCreate(model, data, options, cb);
```

Comment:
I'm not sure if this is the valid logic...
Shouldn't `replaceById` has its own implementation instead of a wrapper method around `replaceOrCreate`? In other words I do not think if `replaceById`should create the instance if it does not exist (that would be the logic for `replaceOrCreate`)